Upgrade to Keycloak 2.5.0

hkeeler opened the pull request:
Closes #47

I've upgrade the Docker image to `2.5.0.Final`.  There were a couple templates that had to be tweaked, and a cache-related change to `standalone.xml`, but that seems to be it!  I've done some local testing, and all appears to be working.

The one wrinkle is that they haven't pushed up a `2.5.0.Final` version of the [Adapters SPI](https://mvnrepository.com/artifact/org.keycloak/keycloak-adapter-spi) Java libs...yet?

I am currently building with `2.4.0.Final`, and again, it _seems_ fine, but I'd like to do a bit more testing before giving the green light to merge.

wpears commented on keycloak/Dockerfile:
```diff
@@ -1,11 +1,12 @@
 ENV PATH=$PATH:/tmp/$MAVEN_DIST/bin
 ENV KC_SPI_SRC=providers
 ENV KC_SPI_DEST=/usr/src/keycloak
+ENV KC_VER=2.4.0.Final
```
What's the story with this vs. 2.5.0. on line 1?

hkeeler replied:
This is the issue I mentioned on https://github.com/cfpb/hmda-platform-auth/pull/49#issue-198604487.  For whatever reason, they never published a `2.5.x` of the Keycloak Java libs.  `2.4.0.Final` libs _seem_ to work fine.

Perhaps I should put in a comment about that, and change the variable name to `KC_LIB_VER`.

awolfe76 commented on keycloak/themes/hmda/login/login.ftl:
```diff
@@ -11,11 +11,11 @@
                         <input id="username" name="username" type="text" autocapitalize="off" autocorrect="off" value="${(login.username!'')?html}" disabled>
                     <#else>
-                        <input id="username" name="username" value="${(login.username!'')?html}" type="text" autofocus />
+                        <input id="username" name="username" value="${(login.username!'')?html}" type="text" autofocus autocomplete="off" />
```
Did it happen to say why `autocomplete="off"`? Seems odd, not sure we need to keep it.

hkeeler replied:
It's a bit strange.  It comes from https://github.com/keycloak/keycloak/commit/5c2bf7c822ec6a8fe72feb52872c452c34df100d, but the corresponding Github PR has no description, and the corresponding JIRA issue doesn't seem to be available to the public.

I can understand why they would make this change though.  Many sites don't allow username fields to be cached.

awolfe76 commented:
@hkeeler - once you add the comments (https://github.com/cfpb/hmda-platform-auth/pull/49#discussion_r94995322) we'll merge.

hkeeler commented:
I updated the envvar in `keycloak/Dockerfile` for clarity, and added a warning comment to `login/theme.properties`.  I did a full _rebuild the world_ with the latest from all repos, and it all works great.  I think we're clear for merge.

awolfe76 commented:
Let me do the same "rebuild the world" and then I'll merge.

awolfe76 commented:
Looks good!
